### PR TITLE
Modify email copy for appellant and recipient when hearing is converted from virtual to video or central

### PR DESCRIPTION
**Pull request by rubaiyat22**
Connects [CASEFLOW-1556](https://vajira.max.gov/browse/CASEFLOW-1556)

### Description
Read [thread](https://dsva.slack.com/archives/C01MBDKAC7P/p1628282082046700?thread_ts=1628004392.010800&cid=C01MBDKAC7P) on why this only implements the copy changes for when virtual hearing is converted back to Video or Central. 
- modify email copy and subject when virtual hearing is converted back to video or central

### Acceptance Criteria
- [x] When a scheduled hearing is converted from Virtual to Video or Central send an email with the following content to the appellant
    - [x] Subject: "Your Board hearing will be held at {Hearing Location or RO Name}"
    - [x] Replace the opening paragraph of the intro section with: "Your hearing with a Veterans Law Judge of the Board of Veterans’ Appeals has changed format. Instead of a virtual tele-hearing, you will go in person to {Hearing Location or RO name} and the Judge will meet with you {via video conference / in person}."
    - [x] Display the following sections underneath the salutation and intro
        - [x] Date and Time
        - [x] Location
        - [x] What should I expect on the day of my hearing?
        - [x] Signature
        - [x] Internal use
- [x] When a scheduled hearing is converted from Virtual to Video or Central and a representative email has already been added to the hearing, send an email with the following content to the representative
    - [x] Subject: "{Full Appellant Name}’s hearing will be held at {Hearing Location or RO Name}"
    - [x] Replace the opening paragraph of the intro section with: "Your client's hearing with a Veterans Law Judge of the Board of Veterans’ Appeals has changed format. Instead of a virtual tele-hearing, you will go in person to {Hearing Location or RO name} and the Judge will meet with you {via video conference / in person}."
    - [x] Display the following sections underneath the salutation and intro
        - [x] Appellant name
        - [x] Case Details link
        - [x] Date and Time
        - [x] Location
        - [x] Signature
        - [x] Internal use

### Testing Plan
```bash
bundle exec rake emails:hearings:sample
Check the content of `cancellation_appellant.rb` and `cancellation_representative.rb`
```

### CODE REVIEW

File: app/views/hearing_mailer/_appellant_location_changed.html.erb [hunk 1]
```diff
@@ -4,23 +4,24 @@
...
+<% end %>
+<% content_for :how_to_join do %>
+  <b>What should I expect on the day of my hearing?</b>
   <p>
-    <%= @virtual_hearing.hearing.hearing_location_or_regional_office&.name %><br/>
-    <%= @virtual_hearing.hearing.hearing_location_or_regional_office&.full_address %><br/>
+    All visitors to VA facilities should bring a photo ID and check in with security 30 minutes prior to the scheduled hearing and report to the front desk. You will be subject to security screening, including the inspection of personal belongings. Masks may be required to enter VA facilities. It is against the law to bring firearms or any other dangerous weapons into a federal facility.
+  </p>
+  <p>
+    Hearings are scheduled in morning and afternoon shifts. The judge will hold each hearing as soon as possible; however, you should be prepared to wait. The approximate time for each hearing is 30 minutes.
```
Comment (tomas-nava) [inline app/views/hearing_mailer/_appellant_location_changed.html.erb:22]: This text is in two locations, would be nice to put it in one place and refer to it from both the reminder and location changed templates.

**Review by tomas-nava — APPROVED** [review]
approve!

with non-blocking comments, and **this one potentially blocking comment below**:

The old subject lines (and all the unchanged subject lines) have a suffix of ` – Do Not Reply`. The new subject lines do not:

| before | after |
| --- | --- |
| Your Board hearing location has changed – Do Not Reply | Your Board hearing will be held at Oakland regional office |

If we want to be consistent with the ` – Do Not Reply`, they should be added to the ends of the new subjects.

**Also!** This PR doesn't resolve [CASEFLOW-1556](https://vajira.max.gov/browse/CASEFLOW-1556), it just connects it (because [CASEFLOW-1556](https://vajira.max.gov/browse/CASEFLOW-1556) includes AC for changes for emails generated with a hearing is converted back to virtual)

File: app/mailers/hearing_mailer.rb [hunk 2]
```diff
@@ -71,6 +72,16 @@ def reminder(email_recipient:, virtual_hearing: nil, hearing: nil)
...
   attr_reader :recipient, :virtual_hearing
 
+  def cancellation_subject
+    # :reek:RepeatedConditionals
+    case recipient.title
+    when HearingEmailRecipient::RECIPIENT_TITLES[:appellant]
+      "Your Board hearing will be held at #{@hearing&.hearing_location_or_regional_office&.name}"
+    when HearingEmailRecipient::RECIPIENT_TITLES[:representative]
+      "#{appellant_name}’s hearing will be held at #{@hearing&.hearing_location_or_regional_office&.name}"
+    end
```
[the remark this answers is not in the document]
Comment (charleyf) [reply app/mailers/hearing_mailer.rb]: After looking more carefully at this I [removed the safe navigation](https://github.com/department-of-veterans-affairs/caseflow/pull/16600/commits/7d72c9ce55a8844c15efe58166aa0cb3dd367cb0).

The [PR](https://github.com/department-of-veterans-affairs/caseflow/pull/14222) and [code for hearing_location_or_regional_office](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/concerns/hearing_location_concern.rb) seem intended to ensure that there's always a location returned.

File: app/mailers/hearing_mailer.rb [hunk 3]
```diff
@@ -108,6 +119,7 @@ def reminder_subject
   end
 
   def confirmation_subject
+    # :reek:RepeatedConditionals
```
[the remark this answers is not in the document]
Comment (charleyf) [reply app/mailers/hearing_mailer.rb:122]: I see how that would combine and simplify the determination of cancellation/confirmation subjects. Thanks Tomas.

**Pull request merged**